solc/CommandLineInterface.cpp

```diff
@@ -672,6 +705,12 @@ void CommandLineInterface::processInput()
 		compile();
 		outputCompilationResults();
 		break;
+	case InputMode::EVMAssemblerJSON:
+		assembleFromEvmAssemblyJson();
+		handleCombinedJSON();
+		handleBytecode("");
+		handleEVMAssembly("");
```

Comment:
This works only by accident - for some reason `CompilerStack` does a weird lookup using unqualified names and ends up accepting `""` as a valid contract name. You should really have the contract name here.

It actually started crashing when I used `EVMAsseblyStack` directly in the branch I mentioned in https://github.com/ethereum/solidity/pull/13673#discussion_r1174181957. I have a fix in that branch.